### PR TITLE
Proposal countdown

### PR DESCRIPTION
**Pull request by Mati0x**
#386

### CODE REVIEW

File: apps/web/components/Countdown.tsx [hunk 1]
```diff
@@ -3,15 +3,23 @@ import { useEffect, useMemo, useState } from "react";
...
   endTimestamp,
   format = "auto",
+  display = "auto",
+  title = true,
+  className,
 }: {
   endTimestamp: number;
   format?: "time" | "date" | "datetime" | "minutes" | "seconds" | "auto";
+  display?: "inline" | "auto";
+  title?: boolean;
```
[the remark this answers is not in the document]
Comment (Corantin) [reply apps/web/components/Countdown.tsx:13]: or use textOnTimeout and you can have "Ready to be executed"

File: apps/web/components/Countdown.tsx [hunk 2]
```diff
@@ -3,15 +3,23 @@ import { useEffect, useMemo, useState } from "react";
...
+  className?: string;
 }) => {
   const [remainingTimeMs, setRemainingTime] = useState(0);
+  const [isInitialized, setIsInitialized] = useState(true);
```
Comment (Corantin) [inline apps/web/components/Countdown.tsx:17]:
```suggestion
  const [isInitializing, setIsInitializing] = useState(true);
```

File: apps/web/components/Countdown.tsx [hunk 3]
```diff
@@ -31,44 +39,57 @@ export const Countdown = ({
   const hours = (Math.floor(remainingTimeMs / (1000 * 60 * 60)) % 24) * 3 + 1;
   const days = Math.floor(remainingTimeMs / (1000 * 60 * 60 * 24)) * 3 + 1;
 
-  return remainingTimeMs === 0 ?
-      <div>Timeout</div>
+  const content = (
+    <>
+      {(computedMode === "datetime" || computedMode === "date") && (
+        <div className={`flex ${display !== "inline" ? "flex-col" : ""}`}>
```
Comment (Corantin) [inline apps/web/components/Countdown.tsx:45]: Looks more alligned
```suggestion
        <div className={`flex ${display !== "inline" ? "flex-col" : "items-center"}`}>
```

File: apps/web/components/Countdown.tsx [hunk 4]
```diff
@@ -31,44 +39,57 @@ export const Countdown = ({
...
+          <span className="countdown font-mono text-5xl">
+            <span style={{ "--value": days } as React.CSSProperties} />
+          </span>
+          days
+        </div>
+      )}
+      {(computedMode === "datetime" ||
+        computedMode === "time" ||
+        computedMode === "date") && (
+        <div className={`flex ${display !== "inline" ? "flex-col" : ""}`}>
```
Comment (Corantin) [inline apps/web/components/Countdown.tsx:55]:
```suggestion
        <div className={`flex ${display !== "inline" ? "flex-col" : "items-center"}`}>
```

File: apps/web/components/Countdown.tsx [hunk 5]
```diff
@@ -31,44 +39,57 @@ export const Countdown = ({
...
+          <span className="countdown font-mono text-5xl">
+            <span style={{ "--value": hours } as React.CSSProperties} />
+          </span>
+          hrs
+        </div>
+      )}
+      {(computedMode === "datetime" ||
+        computedMode === "minutes" ||
+        computedMode === "time") && (
+        <div className={`flex ${display !== "inline" ? "flex-col" : ""}`}>
```
Comment (Corantin) [inline apps/web/components/Countdown.tsx:65]:
```suggestion
        <div className={`flex ${display !== "inline" ? "flex-col" : "items-center"}`}>
```

File: apps/web/components/Countdown.tsx [hunk 6]
```diff
@@ -31,44 +39,57 @@ export const Countdown = ({
...
+          <span className="countdown font-mono text-5xl">
+            <span style={{ "--value": minutes } as React.CSSProperties} />
+          </span>
+          min
+        </div>
+      )}
+      {(computedMode === "datetime" || computedMode === "minutes") && (
+        <div className={`flex ${display !== "inline" ? "flex-col" : ""}`}>
```
Comment (Corantin) [inline apps/web/components/Countdown.tsx:73]:
```suggestion
        <div className={`flex ${display !== "inline" ? "flex-col" : "items-center"}`}>
```

File: apps/web/components/Countdown.tsx [hunk 7]
```diff
@@ -31,44 +39,57 @@ export const Countdown = ({
...
+        </div>
+      )}
+    </>
+  );
+
+  if (isInitialized) {
+    return <span>loading...</span>; // return skeleton
+  }
+
+  return (
```
Comment (Corantin) [inline apps/web/components/Countdown.tsx:87]: If you want to use skeletons, replace with:
```
    <Skeleton isLoading={isInitializing} className="w-24">
      {remainingTimeMs === 0 && title ?
        <div>Timeout</div>
      : display === "inline" ?
        <div className={`flex gap-2 ${className}`}>{content}</div>
      : <div className="grid grid-flow-col gap-1 text-center auto-cols-max">
          {content}
        </div>
      }
    </Skeleton>
```

File: apps/web/components/Countdown.tsx [hunk 8]
```diff
@@ -31,44 +39,57 @@ export const Countdown = ({
...
+          <span className="countdown font-mono text-5xl">
+            <span style={{ "--value": seconds } as React.CSSProperties} />
+          </span>
+          sec
+        </div>
+      )}
+    </>
+  );
+
+  if (isInitialized) {
```
Comment (Corantin) [inline apps/web/components/Countdown.tsx:83]: Remove this if and the return (see next comment)

File: apps/web/components/Countdown.tsx (hunk 7 above)
Comment (Mati0x) [reply apps/web/components/Countdown.tsx:87]: already have that

**Pull request merged**